src/firefly/js/visualize/XYPlotOptions.jsx

```diff
@@ -11,12 +11,37 @@ import {FieldGroup} from '../ui/FieldGroup.jsx';
 import FieldGroupUtils from '../fieldGroup/FieldGroupUtils.js';
 //import InputGroup from '../ui/InputGroup.jsx';
 import Validate from '../util/Validate.js';
+import {Expression} from '../util/expr/Expression.js';
 import {ValidationField} from '../ui/ValidationField.jsx';
 import {CheckboxGroupInputField} from '../ui/CheckboxGroupInputField.jsx';
 import {RadioGroupInputField} from '../ui/RadioGroupInputField.jsx';
 import {SuggestBoxInputField} from '../ui/SuggestBoxInputField.jsx';
 import {FieldGroupCollapsible} from '../ui/panel/CollapsiblePanel.jsx';
 
+import {showInfoPopup} from '../ui/PopupUtil.jsx';
+
+/*
+ * Find the last alphanumeric token in the text
+ * @param {string} text - current content of suggest box
+ * @return {Object} with token and priorContent properties
+ */
+function parseSuggestboxContent(text) {
+    let priorContent = '';
+    let token = '';
+    let priorIdx = -1;
+    var i, c;
+    for (i = text.length-1; i>=0; i--) {
+        c = text.charAt(i);
+        if (!/[A-Za-z\d_]/.test(c)) {
+            priorIdx = i;
+            break;
+        }
+    }
+    if (priorIdx > 0) priorContent = text.substring(0, priorIdx+1);
+    if (priorIdx < text.length) token = text.substring(priorIdx+1);
+
```

Comment:
i am sure this works fine.
just sharing info.. have you tried using string.match?  works great for taking chunks out of matching strings..
i.e.  
> 'cos(x^2)'.match(/(.*)([^A-Za-z\d_])/)
[ 'cos(x^2)', 'cos(x^2', ')', index: 0, input: 'cos(x^2)' ]